doc/users/project/mission.rst

```diff
@@ -0,0 +1,27 @@
+Matplotlib Mission Statement
+============================
+
+The mission of the Matplotlib developer community is to develop, maintain, and
+support Matplotlib and its extensions which provide plotting tools for users of
+the Scientific Python Ecosystem.  We success is based on how effectively we
+enable our users, across scientific, industrial, commercial, and educational
+applications, to achieve their goals.  A diverse developer community creates
+the best software and we welcome anyone, regardless of experience, education,
+gender, or nationality, who shares our mission to the community.
+
+
+Adapting the requirements :ref:`laid out by John Hunter
+<History>` Matplotlib should:
+
+* Support all users of the Scientific Python Ecosystem;
+* Produce high-quality raster and vector formats output;
+* Be embeddedable in graphical user interfaces for application development;
+* Be understandable and extendable by scientists, data scientists, and
```

Comment:
```suggestion
* Be understandable and extendable
```
I'm not sure the scoping is helping here - like would an industry developer (like at Bloomberg) consider themselves an engineer or data scientist?

Reply:
I think yes, most of the people who we have worked with at Bloomberg were in an org with the word "Engineering" in the name.

Fundamentally Matplotlib is a tool for _doing_ science, data science, and engineering and I think it is important that we strive to be accessible to the practitioners of those things.

Reply:
My concern is that folks who do science/data science/engineering/quantitative research methods wouldn't necessarily label themselves as scientists/data scientists/engineers- for example students, who I think are one of our largest user bases and a group we explicitly consider in tutorials/docs/outreach. 

What about separating the two and  tool/task centered framing ? (it still needs wordsmithing, but wanted to jot out something quick)
* Understandable to people^ creating data and scientific data visualizations  
* Extendable by developers who build data and scientific data visualization tools 

^students, researchers, developers, analysts, engineers - I'm not sure what the collective noun for this bucket is

Reply:
I think we can look at the numpy user demographics and use it as a rough guide for us (https://github.com/numpy/numpy-surveys/blob/master/images/2020NumPysurveyresults_community_infographic.pdf).   In the case of numpy, they had 90% of users have bachlor's degrees and only 22% are using the library for "studies" vs 69% who are using it for "work".    Based on this I do not think it is accurate to say students are one of our biggest user bases.

Further students are, by definition, still learning.  It is OK if things are not immediately understandable by students their task to to learn enough so that it is understandable!  I suspect that a majority of the students who are using Matplotlib are doing so because they are in training to be scientists, data scientists, and engineers so I am not worried about saying we are a tool for the role they are in training to do.

I do not see any tension between a) having the goal of being understood by people working with Matplotlib in a professional context and b) making sure our documentation/tutorials do a good job of address people studying to take those roles!

I do not want to split this in half because I think we should be both understandable and extendable by the people who are using Matplotlib to get stuff done.

I stand by the correct collective noun being "scientists, data scientists, and engineers".

Reply:
My prior here is: I think students studying in the sciences are by definition scientists. They are people who are seeking to better understand the world through the application of the scientific method.

I'm willing to be convinced by data or an abundance of opinions to the contrary. I don't think myself 20 years ago would have been turned off by that language, but then again I'm not a part of any group that we struggle to represent here.

Calling out both "scientist" and "data scientist" feels redundant to me, though.

Reply:
> My prior here is: I think students studying in the sciences are by definition scientists
This might be just diss depression, but like I don't really feel like I could call myself a scientist (I cringe at an interview where I did) or an engineer because those aren't official titles I've held. I also am not sure what we gain by including a physicist doing art restoration and excluding an art historian making the exact same heatmaps (and sometimes both on the same interdisciplinary team ). (And yes you may consider the art historian a scientist or data scientist in this context, but I dunno that they would) This is why I was suggesting language that expressed what folks do (create scientific visualizations, build scientific visualization tools) rather than the ways in which they identify.  

> Calling out both "scientist" and "data scientist" feels redundant to me, though.
Yeah, and I don't love it, but the carry w/ 'em distinct notions of the type of analysis folks are doing. And I've been thinking lately about where teachers/teaching track professors would slot in (folks using matplotlib primarily to teach science and math, which is doing science and also is a loaded debate) & I swear I'm not trying to split hairs or be pedantic, but I'm trying to unpack what it is we're really trying to scope for here. 

>I do not see any tension between a) having the goal of being understood by people working with Matplotlib in a professional context and b) making sure our documentation/tutorials do a good job of address people studying to take those roles!

It's not that they're in conflict, it's that the mission statement to me reads like a prioritization of a) in a way that I worry could feel intimidating/like matplotlib is not for b).  Which is why I suggested a split, b/c we want the library to be understandable by a more general audience than we expect to build new tools with it - is why we have a mix of different types of docs.

Reply:
If we are defining who Matplotlib will be comprehensible to, perhaps this could be rephrased based on what level of knowledge we will be try to be comprehensible to, not necessarily domains or job titles. I agree that most scientists and engineers will have that level of knowledge, but no need to exclude a knowledgeable art historian. (Conversely, I do think the mission statement should specifically say we are a scientific visualization library; you are welcome to use it for art, image manipulation, or architecture, but don't ask us to add features tuned for those disciplines as it is out of scope.)

Defining these minimum qualifications is a little harder than tautologically saying that a scientific visualization library should be understandable to scientists, but perhaps more practically useful when making decisions about how in the weeds the docs or API need to go.  Maybe something like "understandable to people with basic python programming skills and familiarity with numerical processing techniques".

Reply:
I think Jody's suggestion or a variation of it is a good alternative.  Suggestion: "understandable to people familiar with data processing techniques and with the Python programming language."  Or "understandable and extendable by..."